src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets

```diff
@@ -1,49 +1,71 @@
 
-  <Target Name="RunPackageValidation" AfterTargets="Pack">
-    
-    <PropertyGroup Condition="'$(ValidateWithPreviousVersion)' == 'true' and '$(PreviousPackagePath)' == ''">
-      <PreviousPackageVersion>$([System.IO.File]::ReadAllText('$(PreviousPackageVersionFilePath)').Trim())</PreviousPackageVersion>
-      <PreviousPackagePath>$(NuGetPackageRoot)$(PreviousPackageName)\$(PreviousPackageVersion)\$(PreviousPackageName).$(PreviousPackageVersion).nupkg</PreviousPackagePath>
+  <Target Name="RunPackageValidation"
+          AfterTargets="Pack">
+    <PropertyGroup Condition="'$(EnablePackageBaselineValidation)' == 'true' and
+                              '$(PackageValidationBaselinePath)' == ''">
+      <PackageValidationBaselineVersion>$([System.IO.File]::ReadAllText('$(PackageValidationIntermediateBaselineFile)').Trim())</PackageValidationBaselineVersion>
+      <PackageValidationBaselinePath>$([MSBuild]::NormalizePath('$(NuGetPackageRoot)', '$(PackageValidationBaselineName)', '$(PackageValidationBaselineVersion)', '$(PackageValidationBaselineName).$(PackageValidationBaselineVersion).nupkg'))</PackageValidationBaselinePath>
     </PropertyGroup>
 
-    <PropertyGroup>
-      <PackagePath Condition="'$(PackagePath)' == ''">$(PackageOutputAbsolutePath)$(PackageId).$(PackageVersion).nupkg</PackagePath>
-    </PropertyGroup>
-
-    <ValidatePackage PackagePath="$(PackagePath)"
-                     PreviousPackagePath="$(PreviousPackagePath)"
+    <!-- PackageTargetPath isn't exposed by NuGet: https://github.com/NuGet/Home/issues/6671. -->
+    <ValidatePackage PackageTargetPath="$([MSBuild]::ValueOrDefault('$(PackageTargetPath)', '$(PackageOutputAbsolutePath)$(PackageId).$(PackageVersion).nupkg'))"
                      RuntimeGraph="$(RuntimeGraph)"
                      NoWarn="$(NoWarn)"
                      RunApiCompat="$([MSBuild]::ValueOrDefault('$(RunApiCompat)', 'true'))"
-                     ValidateWithPreviousVersion="$(ValidateWithPreviousVersion)"/>
+                     BaselineValidation="$(EnablePackageBaselineValidation)"
+                     BaselinePackageTargetPath="$(PackageValidationBaselinePath)" />
   </Target>
 
-  <Target Name="AddPackageDownload" BeforeTargets="CollectPackageDownloads" Condition="'$(ValidateWithPreviousVersion)' == 'true' and '$(PreviousPackagePath)' == ''">
-    <GetLastStablePackage Condition="'$(PreviousPackageVersion)' == ''"
-                          PackageId="$(PreviousPackageName)"
+  <!-- If package validation against a baseline is enabled and a local path to a baseline package
+       isn't specified, try to find the baseline package on the feeds. Achieving this by hooking
+       onto the CollectPackageDownloads extension point in cross-targeting builds. -->
+  <Target Name="AddPackageValidationBaselinePackageAsPackageDownload"
+          BeforeTargets="$(AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets)"
+          DependsOnTargets="_GetRestoreSettings"
```

Comment:
Unfortunately there's no way to avoid the dependency on that internal target. I documented its usage via a code comment. Maybe @nkolev92 has opinions regarding exposing that target as a public one.

Reply:
This is a walk the projects for data target for us. 
It's not something we're using in static graph restore or VS, so I'd be hesitant to expose a public target for that. 
That being said, we're also not likely to remove it any time soon.

Reply:
Hmm as you mention static graph restore, is there a way to get the NuGet sources (`_OutputSources` property) without calling into the target to keep static graph fast and avoid that target to be invoked?

Reply:
Not really. In static graph the reading is done in the code since we can just request the effective value of the properties. 
In regular restore we have to do some tricks to differentiate between a property set in the project file and a property set on the commandline.

Reply:
Got it, thanks. Based on the current limitations around PackageDownload not being able to handle floating versions, we likely need to accept that when using this package and the `EnablePackageBaselineValidation` feature, restore will be slower.